xds/internal/balancer/outlierdetection/subconn_wrapper.go

```diff
@@ -72,3 +94,28 @@ func (scw *subConnWrapper) uneject() {
 func (scw *subConnWrapper) String() string {
 	return fmt.Sprintf("%+v", scw.addresses)
 }
+
+func (scw *subConnWrapper) RegisterHealthListener(listener func(balancer.SubConnState)) {
+	scw.mu.Lock()
+	defer scw.mu.Unlock()
+
+	if !scw.healthListenerEnabled {
+		logger.Errorf("Health listener unexpectedly registered on SubConn %v.", scw)
+		return
+	}
```

Comment:
I think it would be nice to comment somewhere (or quote something from the gRFC) to convey the fact that OD is one health checking mechanism supported by gRPC, just like client-side health checking. Therefore, initially OD LB policy did not enable client-side health checking. But with dualstack support, since pickfirst will be the universal child policy and pickfirst enables client side health checks, OD needs a way to interop with this.